tools/projmgr/src/ProjMgrYamlEmitter.cpp

```diff
@@ -229,6 +235,67 @@ void ProjMgrYamlCbuild::SetComponentFilesNode(YAML::Node node, const ContextItem
   }
 }
 
+void ProjMgrYamlCbuild::SetGeneratorFiles(YAML::Node node, const ContextItem* context, const string& componentId) {
+  if (context->generatorInputFiles.find(componentId) != context->generatorInputFiles.end()) {
+    YAML::Node filesNode;
+    for (const auto& [file, attr, category, version] : context->generatorInputFiles.at(componentId)) {
+      YAML::Node fileNode;
+      SetNodeValue(fileNode[YAML_FILE], FormatPath(file, context->directories.cprj));
+      SetNodeValue(fileNode[YAML_CATEGORY], category);
+      SetNodeValue(fileNode[YAML_ATTR], attr);
+      SetNodeValue(fileNode[YAML_VERSION], version);
+      filesNode.push_back(fileNode);
+    }
+    node[YAML_FILES] = filesNode;
+  }
+}
+
+void ProjMgrYamlCbuild::SetGeneratorsNode(YAML::Node node, const ContextItem* context) {
+  for (const auto& [generatorId, generator] : context->generators) {
+    YAML::Node genNode;
+    const string& workingDir = generator->GetExpandedWorkingDir(context->rteActiveTarget);
+    SetNodeValue(genNode[YAML_WORKING_DIR], FormatPath(workingDir, context->directories.cprj));
+    const string& gpdscFile = generator->GetExpandedGpdsc(context->rteActiveTarget);
+    SetNodeValue(genNode[YAML_GPDSC], FormatPath(gpdscFile, context->directories.cprj));
+
+    const auto& executables = generator->GetExecutables(context->rteActiveTarget);
+    const auto& arguments = generator->GetArguments("exe");
+    for (const string& host : {"win", "linux", "mac", "other"}) {
+      // Executable file
+      YAML::Node commandNode;
+      auto execIt = executables.find(host);
+      if (execIt == executables.end())
+        execIt = executables.find("all");
+      if (execIt == executables.end())
+        continue;
+      commandNode[YAML_FILE] = FormatPath(execIt->second, context->directories.cprj);
+
+      // Arguments
+      YAML::Node argumentsNode;
+      for (const auto& argument : arguments) {
+        if (argument.host == host || argument.host == "all" || argument.host == "") {
+          string arg = argument.argument;
```

Comment:
Maybe the `argument` element of the `GeneratorArgument struct` could be renamed to avoid such pattern `argument.argument`, for example `argument.body`?

Reply:
Sure. But we may not even keep that struct in the end. See discussion with @edriouk above.